src/lib.rs

```diff
@@ -226,7 +226,7 @@ pub enum SchemeData {
 }
 
 /// Components for URLs in a *relative* scheme such as HTTP.
-#[derive(PartialEq, Eq, Clone, Debug, Hash, PartialOrd, Ord)]
+#[derive(Clone, Debug, Hash, PartialOrd, Ord)]
```

Comment:
@eefriedman Before that: "If you are also implementing Eq, there is an additional property that is important:". So there is an invariant to keep in mind when implementing both Eq and Hash. However, Eq and Hash do not need each other, you can just implement one if you want.